src/fiatExchanges/quotes/FiatConnectQuote.ts

```diff
@@ -110,19 +107,39 @@ export default class FiatConnectQuote extends NormalizedQuote {
     return feeString !== undefined ? new BigNumber(feeString) : null
   }
   // FiatConnect quotes denominate fees in fiat & crypto for CashIn & CashOut respectively
-  getFeeInCrypto(exchangeRates: { [token in Currency]: string | null }): BigNumber | null {
+  getFeeInCrypto(
+    exchangeRates: { cGLD: string | null; cUSD: string | null; cEUR: string | null },
+    tokenInfo: TokenBalance | undefined
+  ): BigNumber | null {
+    const fee = this._getFee()
     if (this.flow === CICOFlow.CashOut) {
-      return this._getFee()
+      return fee
+    }
+    const tokenUsdPrice = tokenInfo?.usdPrice
+    const usdExchangeRate = exchangeRates[Currency.Dollar]
+    if (!tokenUsdPrice || !usdExchangeRate || !fee) {
+      return null
     }
-    return convertLocalAmountToCurrency(this._getFee(), exchangeRates[this.getCryptoType()])
+
+    return fee.dividedBy(usdExchangeRate).dividedBy(tokenUsdPrice)
```

Comment:
Seems like this conversion logic is duplicated in ExternalQuote and here. Would be good to just have a single implementation.

Reply:
yep , i joined it with some other logic and added unit tests